onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc

```diff
@@ -400,21 +379,39 @@ Status ResizeOpBuilder::ProcessQDQOpAttrsAndOutputs(QnnModelWrapper& qnn_model_w
   std::vector<std::string> param_tensor_names;
   NodeAttrHelper node_helper(node_unit);
 
-  // Parameter 'exclude_outside'
-  {
-    Qnn_Scalar_t qnn_exclude_outside = QNN_SCALAR_INIT;
-    qnn_exclude_outside.dataType = QNN_DATATYPE_BOOL_8;
-    qnn_exclude_outside.bool8Value = static_cast<uint8_t>(GetOnnxAttr(node_helper, onnx_exclude_outside_attr) != 0);
-
-    QnnParamWrapper qnn_exclude_outside_param(node_unit.Index(), node_unit.Name(), qnn_def::exclude_outside,
-                                              qnn_exclude_outside);
-    param_tensor_names.push_back(qnn_exclude_outside_param.GetParamTensorName());
-    qnn_model_wrapper.AddParamWrapper(std::move(qnn_exclude_outside_param));
-  }
-
-  // Parameter 'transformation_mode'
-  {
-    const std::string transformation_mode = GetOnnxAttr(node_helper, onnx_coord_transf_mode_attr);
+  const std::string interp_mode = GetOnnxAttr(node_helper, onnx_mode_attr);
+  const std::string transformation_mode = GetOnnxAttr(node_helper, onnx_coord_transf_mode_attr);
+  std::string qnn_op_type = "Resize";
+
+  // Handle Resize with {mode: "nearest", coordinate_transformation_mode: "asymmetric"} uniquely.
+  // QNN's ONNX converter tool translates this configuration (regardless of rounding mode)
+  // to QNN's ResizeNearestNeighbor with {align_corners: 0, half_pixel: 0}.
+  //
+  // NOTE: This is only accurate if the rounding mode is "floor". Need to investigate how to handle
+  // other rounding modes with Qualcomm. Ideally, we would use QNN's Resize operator, but it doesn't support
+  // the "asymmetric" coordinate transformation mode on HTP.
+  if (interp_mode == "nearest" && transformation_mode == "asymmetric") {
+    qnn_op_type = "ResizeNearestNeighbor";
```

Comment:
We have different code path for the op type handling, is it possible to make it consistent?

Reply:
We discussed offline and decided to proceed with this approach until we're able to use QNN Resize for all backends.